### ISSUE TITLE
[linux] can't start breach, missing dependencies

### ISSUE THREAD

**Issue by argshook** [spolu/breach_core#92]
I'm trying to try out Breach v0.3.20-alpha on Ubuntu based Linux but to no avail. There are missing dependencies that I'm solving one by one but I still can't get it to run. It might be just me doing silly things and not the Breach itself.

Running `./breach` throws missing `libharfbuzz.so.0` dependency error. I got the pakage from [here (amd64)](http://packages.ubuntu.com/raring/amd64/libharfbuzz0/download). Then `libudev.so.1` was missing, which I got from [here (amd64)](http://packages.ubuntu.com/raring/amd64/libudev1/download) (it depends on `libc6`, which I found [here (amd64)](http://packages.ubuntu.com/raring/amd64/libc6/download). This raised even more dependency problems until I installed libc6:i386 from [here](http://packages.ubuntu.com/raring/i386/libc6/download)).

Running `./breach` at this point didn't raise dependency problems but threw another error: `symbol lookup error: /home/argshook/breach-v0.3.20-alpha.5-linux-x64/__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser: undefined symbol: g_type_class_adjust_private_offset` which is not something I can solve myself.

Updating `libc6:i386` and `libc6:amd64` to v2.17 broke a lot more packages on my machine (but breach didn't throw dependency errors with that version). I had to downgrade them back to v2.15 in order to fix it all up.

I wish I had more time to investigate this further but at this point I'm a bit lost. Any help would greatly help.

**Comment by ssv445** [comment]
+1

**Comment by deepak1556** [comment]
@argshook wats the output of `ldd __AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser` ?

**Comment by argshook** [comment]
```
argshook@MaxPower:~/Manure/breach-v0.3.20-alpha.5-linux-x64$ ldd __AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser
linux-vdso.so.1 =>  (0x00007fff6c551000)
libfreetype.so.6 => /home/argshook/Manure/breach-v0.3.20-alpha.5-linux-x64/__AUTO_UPDATE_BUNDLE__/exo_browser/lib/libfreetype.so.6 (0x00007f2fa6328000)
librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f2fa6101000)
libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f2fa5efd000)
libgtk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0 (0x00007f2fa58c3000)
libgdk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0 (0x00007f2fa5611000)
libatk-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0 (0x00007f2fa53ed000)
libpangocairo-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0 (0x00007f2fa51e1000)
libgdk_pixbuf-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0 (0x00007f2fa4fc1000)
libcairo.so.2 => /usr/lib/x86_64-linux-gnu/libcairo.so.2 (0x00007f2fa4d02000)
libpango-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0 (0x00007f2fa4ab9000)
libfontconfig.so.1 => /usr/lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007f2fa4883000)
libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007f2fa4633000)
libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007f2fa433e000)
libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007f2fa4009000)
libXi.so.6 => /usr/lib/x86_64-linux-gnu/libXi.so.6 (0x00007f2fa3df8000)
libnss3.so => /usr/lib/x86_64-linux-gnu/libnss3.so (0x00007f2fa3aef000)
libnssutil3.so => /usr/lib/x86_64-linux-gnu/libnssutil3.so (0x00007f2fa38c8000)
libsmime3.so => /usr/lib/x86_64-linux-gnu/libsmime3.so (0x00007f2fa36a0000)
libplc4.so => /usr/lib/x86_64-linux-gnu/libplc4.so (0x00007f2fa349b000)
libnspr4.so => /usr/lib/x86_64-linux-gnu/libnspr4.so (0x00007f2fa325c000)
libgconf-2.so.4 => /usr/lib/x86_64-linux-gnu/libgconf-2.so.4 (0x00007f2fa302d000)
libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007f2fa2e03000)
libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007f2fa2bf2000)
libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007f2fa29eb000)
libXrender.so.1 => /usr/lib/x86_64-linux-gnu/libXrender.so.1 (0x00007f2fa27e1000)
libXcomposite.so.1 => /usr/lib/x86_64-linux-gnu/libXcomposite.so.1 (0x00007f2fa25de000)
libasound.so.2 => /usr/lib/x86_64-linux-gnu/libasound.so.2 (0x00007f2fa22f0000)
libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007f2fa20ed000)
libXtst.so.6 => /usr/lib/x86_64-linux-gnu/libXtst.so.6 (0x00007f2fa1ee7000)
libharfbuzz.so.0 => /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007f2fa1c4f000)
libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f2fa1a32000)
libcap.so.2 => /lib/x86_64-linux-gnu/libcap.so.2 (0x00007f2fa182d000)
libudev.so.1 => not found
libdbus-1.so.3 => /lib/x86_64-linux-gnu/libdbus-1.so.3 (0x00007f2fa15e8000)
libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f2fa12e8000)
libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f2fa0feb000)
libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f2fa0dd5000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f2fa0a15000)
/lib64/ld-linux-x86-64.so.2 (0x00007f2fa63aa000)
libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f2fa07fd000)
libgio-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0 (0x00007f2fa04ae000)
libpangoft2-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0 (0x00007f2fa0284000)
libXinerama.so.1 => /usr/lib/x86_64-linux-gnu/libXinerama.so.1 (0x00007f2fa0080000)
libXrandr.so.2 => /usr/lib/x86_64-linux-gnu/libXrandr.so.2 (0x00007f2f9fe78000)
libXcursor.so.1 => /usr/lib/x86_64-linux-gnu/libXcursor.so.1 (0x00007f2f9fc6e000)
libgmodule-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007f2f9fa69000)
libpixman-1.so.0 => /usr/lib/x86_64-linux-gnu/libpixman-1.so.0 (0x00007f2f9f7d2000)
libpng12.so.0 => /lib/x86_64-linux-gnu/libpng12.so.0 (0x00007f2f9f5a9000)
libxcb-shm.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0 (0x00007f2f9f3a6000)
libxcb-render.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-render.so.0 (0x00007f2f9f19c000)
libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007f2f9ef7d000)
libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007f2f9ed75000)
libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007f2f9eb37000)
libplds4.so => /usr/lib/x86_64-linux-gnu/libplds4.so (0x00007f2f9e933000)
libdbus-glib-1.so.2 => /usr/lib/x86_64-linux-gnu/libdbus-glib-1.so.2 (0x00007f2f9e70c000)
libicule.so.48 => /usr/lib/libicule.so.48 (0x00007f2f9e4d7000)
libicuuc.so.48 => /usr/lib/libicuuc.so.48 (0x00007f2f9e16c000)
libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007f2f9df4d000)
libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007f2f9dd30000)
libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007f2f9db2d000)
libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007f2f9d926000)
libicudata.so.48 => /usr/lib/libicudata.so.48 (0x00007f2f9c5b6000)
```

sorry about the `Manure` folder, I keep all uncategorized stuff in there. And since Breach is alpha... _don't hurt me_

**Comment by deepak1556** [comment]
`libudev.so.1` isnt linked. is it in your library path ?

**Comment by argshook** [comment]
This is beyond my knowledge so let me know if I'm doing stupid things. `/lib/x86_64-linux-gnu/` previously did NOT contain `libudev.so.1` ( I had it installed but removed because of broken dependencies `apt-get` was responding).

I have just linked `libudev.so.1` to `libudev.so.0` using `ln -sf /lib/x86_64-linux-gnu/libudev.so.0 /lib/x86_64-linux-gnu/libudev.so.1`. A rather dirty solution, I presume.

Here's `ldd` of `libudev` now:

```
argshook@MaxPower:~/Manure/breach$ ldd __AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser | grep libudev
__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser: /lib/x86_64-linux-gnu/libudev.so.1: no version information available (required by __AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser)
  libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007f1e3e664000)
```

**Comment by iliabylich** [comment]
@argshook  I have same issue

```
➜  breach-v0.3.20-alpha.5-linux-x64  ldd __AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser
    linux-vdso.so.1 =>  (0x00007fffa81fe000)
    libfreetype.so.6 => /home/user/Downloads/breach-v0.3.20-alpha.5-linux-x64/__AUTO_UPDATE_BUNDLE__/exo_browser/lib/libfreetype.so.6 (0x00007fcf7c3d8000)
    librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fcf7c1ac000)
    libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fcf7bfa8000)
    libgtk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0 (0x00007fcf7b96e000)
    libgdk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0 (0x00007fcf7b6bb000)
    libatk-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0 (0x00007fcf7b498000)
    libpangocairo-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0 (0x00007fcf7b28c000)
    libgdk_pixbuf-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0 (0x00007fcf7b06b000)
    libcairo.so.2 => /usr/lib/x86_64-linux-gnu/libcairo.so.2 (0x00007fcf7ad65000)
    libpango-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0 (0x00007fcf7ab1c000)
    libfontconfig.so.1 => /usr/lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007fcf7a8e5000)
    libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007fcf7a696000)
    libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007fcf7a3a1000)
    libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007fcf7a06b000)
    libXi.so.6 => /usr/lib/x86_64-linux-gnu/libXi.so.6 (0x00007fcf79e5b000)
    libnss3.so => /usr/lib/x86_64-linux-gnu/libnss3.so (0x00007fcf79b52000)
    libnssutil3.so => /usr/lib/x86_64-linux-gnu/libnssutil3.so (0x00007fcf7992a000)
    libsmime3.so => /usr/lib/x86_64-linux-gnu/libsmime3.so (0x00007fcf79703000)
    libplc4.so => /usr/lib/x86_64-linux-gnu/libplc4.so (0x00007fcf794fe000)
    libnspr4.so => /usr/lib/x86_64-linux-gnu/libnspr4.so (0x00007fcf792be000)
    libgconf-2.so.4 => /usr/lib/x86_64-linux-gnu/libgconf-2.so.4 (0x00007fcf79090000)
    libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007fcf78e66000)
    libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007fcf78c54000)
    libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007fcf78a4e000)
    libXrender.so.1 => /usr/lib/x86_64-linux-gnu/libXrender.so.1 (0x00007fcf78844000)
    libXcomposite.so.1 => /usr/lib/x86_64-linux-gnu/libXcomposite.so.1 (0x00007fcf78640000)
    libasound.so.2 => /usr/lib/x86_64-linux-gnu/libasound.so.2 (0x00007fcf78353000)
    libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007fcf78150000)
    libXtst.so.6 => /usr/lib/x86_64-linux-gnu/libXtst.so.6 (0x00007fcf77f49000)
    libharfbuzz.so.0 => /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007fcf77cf4000)
    libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fcf77ad6000)
    libcap.so.2 => /lib/x86_64-linux-gnu/libcap.so.2 (0x00007fcf778d0000)
    libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007fcf776bf000)
    libdbus-1.so.3 => /lib/x86_64-linux-gnu/libdbus-1.so.3 (0x00007fcf7747b000)
    libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fcf7717a000)
    libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fcf76e74000)
    libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fcf76c5e000)
    libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fcf76897000)
    /lib64/ld-linux-x86-64.so.2 (0x00007fcf7c45a000)
    libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fcf76680000)
    libgio-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0 (0x00007fcf76331000)
    libpangoft2-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0 (0x00007fcf76106000)
    libXinerama.so.1 => /usr/lib/x86_64-linux-gnu/libXinerama.so.1 (0x00007fcf75f03000)
    libXrandr.so.2 => /usr/lib/x86_64-linux-gnu/libXrandr.so.2 (0x00007fcf75cfb000)
    libXcursor.so.1 => /usr/lib/x86_64-linux-gnu/libXcursor.so.1 (0x00007fcf75af0000)
    libgmodule-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007fcf758ec000)
    libpixman-1.so.0 => /usr/lib/x86_64-linux-gnu/libpixman-1.so.0 (0x00007fcf75654000)
    libpng12.so.0 => /lib/x86_64-linux-gnu/libpng12.so.0 (0x00007fcf7542c000)
    libxcb-shm.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0 (0x00007fcf75229000)
    libxcb-render.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-render.so.0 (0x00007fcf7501e000)
    libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007fcf74e00000)
    libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007fcf74bf7000)
    libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007fcf749ba000)
    libplds4.so => /usr/lib/x86_64-linux-gnu/libplds4.so (0x00007fcf747b6000)
    libdbus-glib-1.so.2 => /usr/lib/x86_64-linux-gnu/libdbus-glib-1.so.2 (0x00007fcf7458f000)
    libgraphite2.so.3 => /usr/lib/x86_64-linux-gnu/libgraphite2.so.3 (0x00007fcf74373000)
    libcgmanager.so.0 => /lib/x86_64-linux-gnu/libcgmanager.so.0 (0x00007fcf74158000)
    libnih.so.1 => /lib/x86_64-linux-gnu/libnih.so.1 (0x00007fcf73f40000)
    libnih-dbus.so.1 => /lib/x86_64-linux-gnu/libnih-dbus.so.1 (0x00007fcf73d36000)
    libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007fcf73b16000)
    libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fcf738fb000)
    libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007fcf736f7000)
    libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007fcf734f1000)
```

But seems in my case `libudev.so.1` is linked.

```
➜  breach-v0.3.20-alpha.5-linux-x64  ./breach                                                              
/home/user/Downloads/breach-v0.3.20-alpha.5-linux-x64/__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser: symbol lookup error: /home/user/Downloads/breach-v0.3.20-alpha.5-linux-x64/__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser: undefined symbol: g_type_class_adjust_private_offset
```

**Comment by deepak1556** [comment]
@iliabylich i guess you might have an outdated `glib`, can you try updating and see if it helps ?

**Comment by oren** [comment]
I am getting error when trying to install the binary on ubuntu 14.04, 64bit.
Here are the commands I am running and their output:

./breach

```
[2566:2566:0712/005047:97712961665:FATAL:browser_main_loop.cc(172)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
./breach: line 6:  2566 Aborted                 (core dumped) $SRC_DIR/__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser --raw $SRC_DIR/__AUTO_UPDATE_BUNDLE__/breach_core
```

./breach --no-sandbox

```
[2566:2566:0712/005047:97712961665:FATAL:browser_main_loop.cc(172)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
./breach: line 6:  2566 Aborted                 (core dumped) $SRC_DIR/__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser --raw $SRC_DIR/__AUTO_UPDATE_BUNDLE__/breach_core
```

./**AUTO_UPDATE_BUNDLE**/exo_browser/exo_browser --raw ./**AUTO_UPDATE_BUNDLE**/breach_core --no-sandbox

```
./__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser: bad option: --no-sandbox
~/Downloads/breach-v0.3.20-alpha.5-linux-x64 [4502:4502:0100/000000:100653340720:ERROR:zygote_linux.cc(546)] write: Broken pipe
```

ldd **AUTO_UPDATE_BUNDLE**/exo_browser/exo_browser

```
linux-vdso.so.1 =>  (0x00007fffbc3a9000)
libfreetype.so.6 => /home/oren/Downloads/breach-v0.3.20-alpha.5-linux-x64/__AUTO_UPDATE_BUNDLE__/exo_browser/lib/libfreetype.so.6 (0x00007fb6a9d87000)
librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fb6a9b66000)
libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fb6a9962000)
libgtk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgtk-x11-2.0.so.0 (0x00007fb6a9326000)
libgdk-x11-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk-x11-2.0.so.0 (0x00007fb6a9073000)
libatk-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libatk-1.0.so.0 (0x00007fb6a8e51000)
libpangocairo-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangocairo-1.0.so.0 (0x00007fb6a8c44000)
libgdk_pixbuf-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgdk_pixbuf-2.0.so.0 (0x00007fb6a8a22000)
libcairo.so.2 => /usr/lib/x86_64-linux-gnu/libcairo.so.2 (0x00007fb6a8717000)
libpango-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0 (0x00007fb6a84ca000)
libfontconfig.so.1 => /usr/lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007fb6a828d000)
libgobject-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgobject-2.0.so.0 (0x00007fb6a803c000)
libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007fb6a7d34000)
libX11.so.6 => /usr/lib/x86_64-linux-gnu/libX11.so.6 (0x00007fb6a79fe000)
libXi.so.6 => /usr/lib/x86_64-linux-gnu/libXi.so.6 (0x00007fb6a77ee000)
libnss3.so => /usr/lib/x86_64-linux-gnu/libnss3.so (0x00007fb6a74b5000)
libnssutil3.so => /usr/lib/x86_64-linux-gnu/libnssutil3.so (0x00007fb6a7289000)
libsmime3.so => /usr/lib/x86_64-linux-gnu/libsmime3.so (0x00007fb6a705d000)
libplc4.so => /usr/lib/x86_64-linux-gnu/libplc4.so (0x00007fb6a6e58000)
libnspr4.so => /usr/lib/x86_64-linux-gnu/libnspr4.so (0x00007fb6a6c1a000)
libgconf-2.so.4 => /usr/lib/x86_64-linux-gnu/libgconf-2.so.4 (0x00007fb6a69eb000)
libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007fb6a67c1000)
libXext.so.6 => /usr/lib/x86_64-linux-gnu/libXext.so.6 (0x00007fb6a65ae000)
libXfixes.so.3 => /usr/lib/x86_64-linux-gnu/libXfixes.so.3 (0x00007fb6a63a8000)
libXrender.so.1 => /usr/lib/x86_64-linux-gnu/libXrender.so.1 (0x00007fb6a619e000)
libXcomposite.so.1 => /usr/lib/x86_64-linux-gnu/libXcomposite.so.1 (0x00007fb6a5f9a000)
libasound.so.2 => /usr/lib/x86_64-linux-gnu/libasound.so.2 (0x00007fb6a5caa000)
libXdamage.so.1 => /usr/lib/x86_64-linux-gnu/libXdamage.so.1 (0x00007fb6a5aa7000)
libXtst.so.6 => /usr/lib/x86_64-linux-gnu/libXtst.so.6 (0x00007fb6a58a0000)
libharfbuzz.so.0 => /usr/lib/x86_64-linux-gnu/libharfbuzz.so.0 (0x00007fb6a564b000)
libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fb6a542d000)
libcap.so.2 => /lib/x86_64-linux-gnu/libcap.so.2 (0x00007fb6a5227000)
libudev.so.1 => /lib/x86_64-linux-gnu/libudev.so.1 (0x00007fb6a5016000)
libdbus-1.so.3 => /lib/x86_64-linux-gnu/libdbus-1.so.3 (0x00007fb6a4dd1000)
libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fb6a4acc000)
libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fb6a47c6000)
libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fb6a45b0000)
libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fb6a41e9000)
/lib64/ld-linux-x86-64.so.2 (0x00007fb6a9e09000)
libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007fb6a3fd0000)
libgmodule-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgmodule-2.0.so.0 (0x00007fb6a3dcc000)
libgio-2.0.so.0 => /usr/lib/x86_64-linux-gnu/libgio-2.0.so.0 (0x00007fb6a3a58000)
libpangoft2-1.0.so.0 => /usr/lib/x86_64-linux-gnu/libpangoft2-1.0.so.0 (0x00007fb6a3843000)
libXinerama.so.1 => /usr/lib/x86_64-linux-gnu/libXinerama.so.1 (0x00007fb6a363f000)
libXrandr.so.2 => /usr/lib/x86_64-linux-gnu/libXrandr.so.2 (0x00007fb6a3435000)
libXcursor.so.1 => /usr/lib/x86_64-linux-gnu/libXcursor.so.1 (0x00007fb6a322b000)
libpixman-1.so.0 => /usr/lib/x86_64-linux-gnu/libpixman-1.so.0 (0x00007fb6a2f81000)
libpng12.so.0 => /lib/x86_64-linux-gnu/libpng12.so.0 (0x00007fb6a2d5b000)
libxcb-shm.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-shm.so.0 (0x00007fb6a2b58000)
libxcb-render.so.0 => /usr/lib/x86_64-linux-gnu/libxcb-render.so.0 (0x00007fb6a294e000)
libxcb.so.1 => /usr/lib/x86_64-linux-gnu/libxcb.so.1 (0x00007fb6a272f000)
libthai.so.0 => /usr/lib/x86_64-linux-gnu/libthai.so.0 (0x00007fb6a2526000)
libffi.so.6 => /usr/lib/x86_64-linux-gnu/libffi.so.6 (0x00007fb6a231d000)
libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007fb6a20df000)
libplds4.so => /usr/lib/x86_64-linux-gnu/libplds4.so (0x00007fb6a1eda000)
libdbus-glib-1.so.2 => /usr/lib/x86_64-linux-gnu/libdbus-glib-1.so.2 (0x00007fb6a1cb3000)
libgraphite2.so.3 => /usr/lib/x86_64-linux-gnu/libgraphite2.so.3 (0x00007fb6a1a96000)
libcgmanager.so.0 => /lib/x86_64-linux-gnu/libcgmanager.so.0 (0x00007fb6a187b000)
libnih.so.1 => /lib/x86_64-linux-gnu/libnih.so.1 (0x00007fb6a1663000)
libnih-dbus.so.1 => /lib/x86_64-linux-gnu/libnih-dbus.so.1 (0x00007fb6a1458000)
libselinux.so.1 => /lib/x86_64-linux-gnu/libselinux.so.1 (0x00007fb6a1235000)
libresolv.so.2 => /lib/x86_64-linux-gnu/libresolv.so.2 (0x00007fb6a1019000)
libXau.so.6 => /usr/lib/x86_64-linux-gnu/libXau.so.6 (0x00007fb6a0e15000)
libXdmcp.so.6 => /usr/lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007fb6a0c0e000)
libdatrie.so.1 => /usr/lib/x86_64-linux-gnu/libdatrie.so.1 (0x00007fb6a0a07000)
```

**Comment by deepak1556** [comment]
@oren have you tried this https://github.com/breach/breach_core/issues/60#issuecomment-48656278

**Comment by oren** [comment]
I just tried it. works! 
i see a 'welcome' page. thank you.

**Comment by spolu** [comment]
Hi guys,
I've made a build with Ubuntu 12.04 of the exobrowser. Can you try running it and tell me it that solves these problems: https://github.com/breach/releases/tree/master/exo_browser/v0.6.1916

**Comment by argshook** [comment]
@spolu it didn't work for me, I was getting this error:

```
argshook@MaxPower:~/Manure/breach$ ./breach 
[25189:25189:0717/001129:61520711587:FATAL:browser_main_loop.cc(172)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on developing with the sandbox on.
./breach: line 6: 25189 Aborted                 $SRC_DIR/__AUTO_UPDATE_BUNDLE__/exo_browser/exo_browser --raw $SRC_DIR/__AUTO_UPDATE_BUNDLE__/breach_core
```

Although [#60 comment](https://github.com/breach/breach_core/issues/60#issuecomment-48656278) did work for me. Breach is opened and adding modules as I'm writing this.

**Comment by spolu** [comment]
So all the glib related errors are gone! That's great :)

**Comment by spolu** [comment]
This should be solved by the release of `v0.3.22-alpha.6`
Linux x64: http://bit.ly/1p8Jkmc

**Comment by spolu** [comment]
And make sure to checkout README in tarball or: https://github.com/breach/breach_core/wiki/Running-Breach-on-Linux
